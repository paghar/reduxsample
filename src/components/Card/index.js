import React from "react"
import StarRating from "../StarRating"
import {
  CardWrapper,
  CardContent,
  CardImage,
  CardTitle,
  CardText,
  RankWrapper
} from "./index.style"
const Card = ({ image, title, context, rateValue, clicked }) => {
  return (
    <CardWrapper>
      <CardContent>
        <CardImage src={image} />
        <CardTitle>{title}</CardTitle>
        <CardText>{context}</CardText>
        <RankWrapper>
          <StarRating totalStars={5} rateValue={rateValue} clicked={clicked} />
        </RankWrapper>
      </CardContent>
    </CardWrapper>
  )
}

export default Card
