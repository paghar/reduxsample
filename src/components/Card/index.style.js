import styled from "styled-components"

export const CardWrapper = styled.div`
  background-color: white;
  border-radius: 0.25rem;
  box-shadow: 0 20px 40px -14px rgba(0, 0, 0, 0.25);
  display: flex;
  flex-direction: column;
  overflow: hidden;
`

export const CardContent = styled.div`
  background: linear-gradient(to bottom left, #ef8d9c 40%, #ffc39e 100%);
`

export const CardTitle = styled.h2`
  color: #ffffff;
  font-size: 1.1rem;
  font-weight: 700;
  letter-spacing: 1px;
  text-transform: capitalize;
  margin-left: 1rem;
`
export const CardImage = styled.img`
  height: auto;
  max-width: 100%;
  vertical-align: middle;
`

export const CardText = styled.p`
  color: #ffffff;
  font-size: 0.875rem;
  line-height: 1.5;
  margin: 0rem 0.2rem 1.25rem 1rem;
  font-weight: 400;
`
export const RankWrapper = styled.div`
  color: #ffffff;
  padding: 0.8rem;
  border-radius: 4px;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid rgba(255, 255, 255, 0.2);
  background: transparent;
  background-color: rgba(255, 255, 255, 0.12);
  margin: 0rem 0.5rem 1.25rem 0.5rem;
`
