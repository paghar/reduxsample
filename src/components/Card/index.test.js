import React from "react"
import renderer from "react-test-renderer"
import Card from "."
import { mount } from "enzyme"
import Image1 from "../../assets/image/1.jpg"
describe("<Card />", () => {
  it("Card", () => {
    const tree = renderer
      .create(
        <Card
          image={Image1}
          title="Chicken fried steak"
          context="chicken fried steak was born"
          rateValue="1"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Card Click Event", () => {
    const mockCallBack = jest.fn()
    const wrapper = mount(
      <Card
        image={Image1}
        title="Chicken fried steak"
        context="chicken fried steak was born"
        rateValue="1"
        clicked={mockCallBack}
      />
    )

    wrapper
      .find("div")
      .at(5)
      .simulate("click")
    expect(mockCallBack).toHaveBeenCalled()
  })
})
