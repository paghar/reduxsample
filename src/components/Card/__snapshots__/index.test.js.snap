// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Card 1`] = `
<div
  className="sc-bdVaJa bZiTtu"
>
  <div
    className="sc-bwzfXH eoQqZU"
  >
    <img
      className="sc-bxivhb eJNyue"
      src="1.jpg"
    />
    <h2
      className="sc-htpNat icPpiZ"
    >
      Chicken fried steak
    </h2>
    <p
      className="sc-ifAKCX gpqyoM"
    >
      chicken fried steak was born
    </p>
    <div
      className="sc-EHOje cTTRho"
    >
      <div
        className="star-rating"
      >
        <div
          className="star selected"
          onClick={[Function]}
        />
        <div
          className="star"
          onClick={[Function]}
        />
        <div
          className="star"
          onClick={[Function]}
        />
        <div
          className="star"
          onClick={[Function]}
        />
        <div
          className="star"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
